sqlglot/transforms.py

```diff
@@ -79,6 +78,36 @@ def eliminate_distinct_on(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def eliminate_qualify(expression: exp.Expression) -> exp.Expression:
+    """
+    Convert SELECT statements that contain the QUALIFY clause into subqueries, filtered equivalently.
+
+    The idea behind this transformation can be seen in Snowflake's documentation for QUALIFY:
+    https://docs.snowflake.com/en/sql-reference/constructs/qualify
+
+    Some dialects don't support window functions in the WHERE clause, so we need to include them as
+    projections in the subquery, in order to refer to them in the outer filter using aliases. Also,
+    if a column is referenced in the QUALIFY clause but is not selected, we need to include it too,
+    otherwise we won't be able to refer to it in the outer query's WHERE clause.
+    """
+    if isinstance(expression, exp.Select) and expression.args.get("qualify"):
+        outer_selects = exp.select(*[s.alias_or_name or s for s in expression.selects])
```

Comment:
this is not right, you should put a test case for this but you need to create an alias if it doesn't have one

imagine you have something like select a + 1, you're going to then re-select a + 1 in the outer select but a won't exist.

Reply:
Good catch, thanks. Let me know if I missed something in my implementation.